src/main/scala/model/Gene.scala

```diff
@@ -0,0 +1,62 @@
+package model
+
+import model.Alleles.AlleleKind
+import model.Genes.GeneKind
+import model.GenesUtils.getGeneKind
+
+/**
+ * Represents an Allele of a Gene of a specific Bunny.
+ */
+sealed trait Allele {
+  val kind: AlleleKind
+  val isMutated: Boolean
+  def getCaseSensitiveLetter(letter: String): String = {
+    if (kind.isDominant.isDefined) {
+      if (kind.isDominant.get) letter.toUpperCase else letter.toLowerCase
+    } else ""
+  }
+}
+
+/**
+ * Represents a standard allele.
+ * @param kind the kind of the Allele.
+ */
+case class StandardAllele(kind: AlleleKind) extends Allele {
+  override val isMutated: Boolean = true
+}
+
+/**
+ * Represents an allele which has just been mutated.
+ * @param kind the kind of the Allele.
+ */
+case class JustMutatedAllele(kind:AlleleKind) extends Allele{
+  override val isMutated: Boolean = false
+  if (getGeneKind(kind).mutated != kind) throw new InconsistentMutatedAlleleException
+}
+
+/**
+ * Represents a Gene of a specific Bunny.
+ */
+trait Gene {
+  val kind: GeneKind
+  val momAllele: Allele
+  val dadAllele: Allele
+
+  def getVisibleTrait: AlleleKind =
+    if (momAllele.kind == dadAllele.kind || momAllele.kind.isDominant.getOrElse(false)) momAllele.kind else dadAllele.kind
+  def getLetters: String = momAllele.getCaseSensitiveLetter(kind.letter) + dadAllele.getCaseSensitiveLetter(kind.letter)
+
+  if (!(momAllele.kind == kind.mutated || momAllele.kind == kind.base)
+    && (dadAllele.kind == kind.mutated || dadAllele.kind == kind.base))
+    throw new InconsistentAlleleException
+}
+
+/**
+ *  Represents a Standard Gene of a specific Bunny.
+ * @param kind      the kind of Gene
+ * @param momAllele the allele from the mom
+ * @param dadAllele the allele from the dad
+ */
+case class StandardGene(kind: GeneKind,
```

Comment:
In una futura implementazione credi esisterà un gene non standard?

Reply:
No, però volendo fare l'interfaccia separata (il trait), non me li fa chiamare allo stesso modo. O eliminiamo l'interfaccia o ci teniamo parte della classi con "Standard" (o "Base" o quel che vogliamo)

Reply:
Io proporrei di fare un companion object "Gene" che nel metodo apply istanzia l'oggetto della classe che implementa il trait (GeneImpl) e lato utilizzatore si usa Gene

Reply:
Ripensandoci, noi non avremo Gene non standard, ma penso che in natura ci possano essere, ad esempio il caso in cui nell'eterozigote fa vedere parte entrambi gli Alleli (fiore rosa)

Reply:
Okay però quella è più una cosa a livello di fenotipo, appena riesco comunque ti propongo la mia soluzione che sicuro non mi sono spiegata bene :)

Reply:
Io stavo pensando ad una cosa del genere:
`trait Gene {
  val kind: GeneKind
  val momAllele: Allele
  val dadAllele: Allele

  def getVisibleTrait: AlleleKind =
    if (momAllele.kind == dadAllele.kind || momAllele.kind.isDominant.getOrElse(false)) momAllele.kind else dadAllele.kind
  def getLetters: String = momAllele.getCaseSensitiveLetter(kind.letter) + dadAllele.getCaseSensitiveLetter(kind.letter)
}



object Gene {
  def apply(kind: GeneKind, momAllele: Allele, dadAllele: Allele): Gene = {
    if (!(momAllele.kind == kind.mutated || momAllele.kind == kind.base)
      && (dadAllele.kind == kind.mutated || dadAllele.kind == kind.base))
      throw new InconsistentAlleleException
    new GeneImpl(kind, momAllele, dadAllele)
  }

  private class GeneImpl(kind: GeneKind,
                          momAllele: Allele,
                          dadAllele: Allele) extends Gene
}`